### ISSUE TITLE
How to configure aXe with mix of WCAG 2.0/2.1 A/AA/AAA and Rules

### ISSUE THREAD

**Issue by yhtang68** [dequelabs/axe-core#1723]
Hello,
I have further questions on this thread (https://github.com/dequelabs/axe-core/issues/261)

1) How could we indicate WCAG 2.0 A/AA Rules and WCAG 2.1 A/AA Rules together? Could we document this usage in (https://www.deque.com/axe/axe-for-web/documentation/api-documentation/)
e.g. I like to have
WCAG 2.0 Guidelines
1.1.1 Non-text Content - Level A
1.2.1 Audio-only and Video-only (Prerecorded) - Level A
1.2.2 Captions (Prerecorded) - Level A
3.1.2 Language of Parts - Level AA
3.2.1 On Focus - Level A
3.2.2 On Input - Level A
3.2.3 Consistent Navigation - Level AA
3.2.4 Consistent Identification - Level AA
AND
WCAG 2.1 Guidelines
1.3.4 Orientation - Level AA
1.3.5 Identify Input Purpose - Level AA
1.4.10 Reflow - Level AA
1.4.11 Non-Text Contrast - Level AA

2) What WCAG Rules are currently can NOT be tested via aXe? that required to be manually tested, e.g. Screen Reader?

3) How we may apply the rules defined in 1) when using aXe browser plugin? (https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US)

Thank you
Andy

**Comment by straker** [comment]
In answer to your first question, you can use the WCAG success criterion number as a tag and run any rule that we've designated to that SC.

```js
await axe.run({
  runOnly: {
    type: "tag",
    values: ["wcag111", "wcag121"]
  }
});
```

You can see which rules we have that will run for each of the WCAG success criterion in [doc/rule-descriptions.md](https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md).

@jeankaplansky this doesn't seem to be listed in https://www.deque.com/axe/axe-for-web/documentation/api-documentation/ in any section we list the available tags, could we add it?

I don't know which WCAG rules we don't test. Maybe @WilcoFiers or @dylanb can answer that.

I don't believe you can apply a custom subset of rules to run in the browser extension. I believe it just runs all the non-experimental rules we have (Wilco or Dylan please correct me if I'm wrong).

**Comment by yhtang68** [comment]
Hi Steven,
**RE: Q 1) you can use the WCAG success criterion number as a tag and run any rule that we've designated to that SC.**
Given example didn't show how we have a mixed of WCAG Version: 2.0/2.1 and Level A/AA/AAA but only the rules wcag111, wcag 121.
Could you please show me a mixed example for the set I listed above? 

**RE:  Q 3) I don't believe you can apply a custom subset of rules to run in the browser extension. I believe it just runs all the non-experimental rules we have.**
So what are the non-experimental rules?


Thank you

**Comment by WilcoFiers** [comment]
1. I don't fully understand the question. Can you clarify what you are looking to do? I would think Steve answered your question as best I can though. If you want all rules for a particular WCAG level, use the level tags. If you want a subset, use the tag for the WCAG SC. If you want specific rules, run with `type: 'rule'` and list the rule IDs

2. Axe-core does not test any of the success criteria completely. You should always consider doing additional manual testing if you have the opportunity to do so.

3. Only the rules without the "experimental" flag, and without `enabled: false` on them run in the axe extension.

**Comment by yhtang68** [comment]
1) I see what you're saying now. I have some misunderstand of WCAG Version, Level and Rule i.e. SC.
Now I understand SC indicate WCAG 2.0/2.1 and A, AA, AAA already. So in my case I only need to indicate SC.
```
await axe.run({
  runOnly: {
    type: "tag",
    values: ["wcag111", "wcag121"]
  }
});
```
2) It's answered here (https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md)

3) Could you explain more here. How to get the SC (that are without the "experimental" flag, and without enabled: false)?

Thank you

**Comment by straker** [comment]
the https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md list has the tags for each rule and if it's enabled by default or not. So for example, the rule `focus-order-semantics` is an experimental rule while the `audio-caption` rule is not enabled. Neither of these two rules will be run.

**Comment by TheJaredWilcurt** [comment]
I searched for "Configure axe-core to use AAA" and it took me here. Leaving this as a note for others.

Here's what I finally got to work:

```js
// axe-core v4.10.3 at time of writing
import axe from 'axe-core';

axe.configure({
  rules: [
    // Enable AA contrast violations for colors with contrast ratio below 4.5:1
    {
      id: 'color-contrast',
      enabled: true
    },
    // Enable AAA contrast violations for colors with contrast ratio between 4.5:1 and 7:1
    {
      id: 'color-contrast-enhanced',
      enabled: true
    }
  ]
});

axe
  .run(window.document)
  .then(({ violations }) => {
    console.log({ violations });
  })
  .catch((error) => {
    console.log({ error });
  });
```
